docs/contributor/commits_pull_requests_and_merging.md

```diff
@@ -0,0 +1,108 @@
+* `fix:` - When fixing a bug, we should use this type. Commits using this type will always show up in the Changelog.
+* `chore:` - The least used type, these are **not** included in the Changelog unless they are breaking changes.
+
+We've used some legacy types previously, but these often have a better format:
+
+* `docs:` - These should be annotated as `chore(docs)` if they aren't being changed with a feature or fix.
+* `refactor:` - These should be annotated as `chore(refactor)`.
+* `ci:` - These should be annotated as `chore(ci)`.
+
+### Conventional Commits: Breaking Changes
+
+Annotating **BREAKING CHANGES** is extremely important to our release process and versioning. To mark a commit as breaking, we add the `!` character after the type, but before the colon. For example:
+
+```
+feat!: Update assignment semantics (#560)
+```
+
+### Conventional Commits: Scopes
+
+In the past, we weren't very disciplined about using scopes; however, they significantly improve the Changelog, so we want to use a scope whenever possible. If we are only changing one part of the project, we can use the name of the workspace, like `(stdlib)` or `(grainc)`. If a change touches multiple parts of the codebase, there might be a better scope, like `(refactor)`.
+
+```
+feat(stdlib): Add Array.reverse() function (#698)
+```
+
+```
+fix(grainc): Apply separator normalization before cmdliner util (#654)
+```
+
+## Pull Requests
+
+Pull requests should be focused on the specific change they are working towards. If additional work is required to finish the original pull request, that work should be submitted as a separate one.
+
+This strategy avoids scenarios where pull requests grow too large/out-of-scope and don't get proper reviews—we want to avoid "LGTM, I trust you" reviews.
+
+The easiest way to do this is to have multiple Conventional Commits while you work and then you can cherry-pick the smaller changes into separate branches for pull requesting.
+
+### Reviews
+
+As the Grain core team has grown, we've adjusted our code review process. Currently, we require **two** approvals from core team members for any non-breaking pull request and **three** approvals for pull requests containing breaking changes.
+
+### With Breaking Changes
+
+Sometimes, we don't merge pull requests with breaking changes immediately upon approval. Since a breaking change will cause Grain to bump to the next "minor" version, we might want to land some fixes in "patch" releases before we begin working on that next breaking version.
+
+## Merging
+
+Once approved by the required number of core team members, the submitter can merge their changes into the `main` branch. Sometimes, especially for external contributions, the final approver may merge the pull request instead of the submitter.
+
+We use "squash merging" to merge pull requests, this will cause all commits to be combined into one commit—another reason we want to keep pull requests small & focused.
```

Comment:
```suggestion
We use "squash merging" to merge pull requests. This will cause all commits to be combined into one commit—another reason we want to keep pull requests small & focused.
```